.circleci/config.yml

```diff
@@ -15,7 +15,7 @@ jobs:
       - run:
           name: setup
           command: |
-              sudo pip install -U black flake8 "isort[pyproject]" numpy pytest sphinx pillow tqdm hypothesis
+              sudo pip install -U black flake8 flake8-bugbear flake8-comprehensions "isort[pyproject]" numpy pytest sphinx pillow tqdm hypothesis
```

Comment:
`flake8-comprehensions` looks a little too opinionated and annoying IMO.  For instance, I personally prefer `set(...)` instead of `{...}` as the latter is too visually similar to a list or dict comprehension despite having very different implications, which makes scanning code hard.

My vote is to keep the linter to looking for things that are actual bugs, not small performance optimizations -- especially given that for anything we actually cared about the performance of, we'd write in cpp.

Reply:
@erikwijmans I can add that specific rule to the ignore, but some of the other ones are valuable.

Reply:
@erikwijmans Yeah, PyTorch has C408 on its global ignore list so I'll add it too.

Reply:
@erikwijmans Also I agree about disabling bits that affect readability, but a lot of flake8-comprehensions are performance optimizations that make a lot of sense and don't compromise readability (like not converting a generator to a list before its fed to a builtin that takes an iterator). 

Since we do care about speed, I would vote on keeping it, but adding any of the opinionated errors codes to the flake8 ignore config.